server/service/apple_mdm.go

```diff
@@ -3903,11 +3903,16 @@ func NewInstalledApplicationListResultsHandler(
 					return ctxerr.Wrap(ctx, err, "request refetch for host after vpp install verification")
 				}
 			default:
-				if _, err := ds.UpdateHostSoftware(ctx, hostID, installedAppResult.AvailableApps()); err != nil {
-					return ctxerr.Wrap(ctx, err, "InstalledApplicationList handler: update host software")
+				err = commander.InstalledApplicationList(ctx, []string{installedAppResult.HostUUID()}, fleet.RefetchAppsCommandUUID(), false)
+				if err != nil {
+					return ctxerr.Wrap(ctx, err, "refetch apps with MDM")
 				}
-			}
 
+				err = ds.AddHostMDMCommands(ctx, []fleet.HostMDMCommand{{HostID: hostID, CommandType: fleet.RefetchAppsCommandUUIDPrefix}})
```

Comment:
I had a slight feeling that this should be in a DB txn, with the DB call first, so if the DB call fails the command doesn't go out, but we're not actually assuming that we're the only ones ever to send commands to devices I think, so the failure mode of not being able to insert after the command has already been queued is probably less disruptive, and throwing this inside a transaction would be rather disruptive anyway from a refactor perspective, so fair enough.